source/credentials_provider_sts_web_identity.c

```diff
@@ -0,0 +1,1120 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/auth/credentials.h>
+
+#include <aws/auth/external/cJSON.h>
+#include <aws/auth/private/aws_profile.h>
+#include <aws/auth/private/credentials_utils.h>
+#include <aws/auth/private/xml_parser.h>
+#include <aws/common/clock.h>
+#include <aws/common/date_time.h>
+#include <aws/common/environment.h>
+#include <aws/common/string.h>
+#include <aws/common/uuid.h>
+#include <aws/http/connection.h>
+#include <aws/http/connection_manager.h>
+#include <aws/http/request_response.h>
+#include <aws/http/status_code.h>
+#include <aws/io/file_utils.h>
+#include <aws/io/logging.h>
+#include <aws/io/socket.h>
+#include <aws/io/tls_channel_handler.h>
+#include <aws/io/uri.h>
+
+#if defined(_MSC_VER)
+#    pragma warning(disable : 4204)
+#endif /* _MSC_VER */
+
+#define STS_WEB_IDENTITY_RESPONSE_SIZE_INITIAL 2048
+#define STS_WEB_IDENTITY_RESPONSE_SIZE_LIMIT 10000
+#define STS_WEB_IDENTITY_CONNECT_TIMEOUT_DEFAULT_IN_SECONDS 2
+#define STS_WEB_IDENTITY_CREDS_DEFAULT_DURATION_SECONDS 900
+#define STS_WEB_IDENTITY_MAX_ATTEMPTS 3
+
+struct aws_credentials_provider_sts_web_identity_impl {
+    struct aws_http_connection_manager *connection_manager;
+    struct aws_credentials_provider_system_vtable *function_table;
+    struct aws_string *role_arn;
+    struct aws_string *role_session_name;
+    struct aws_string *token_file_path;
+    struct aws_tls_ctx *ctx;
+    struct aws_tls_connection_options connection_options;
+};
+
+static struct aws_credentials_provider_system_vtable s_default_function_table = {
+    .aws_http_connection_manager_new = aws_http_connection_manager_new,
+    .aws_http_connection_manager_release = aws_http_connection_manager_release,
+    .aws_http_connection_manager_acquire_connection = aws_http_connection_manager_acquire_connection,
+    .aws_http_connection_manager_release_connection = aws_http_connection_manager_release_connection,
+    .aws_http_connection_make_request = aws_http_connection_make_request,
+    .aws_http_stream_activate = aws_http_stream_activate,
+    .aws_http_stream_get_connection = aws_http_stream_get_connection,
+    .aws_http_stream_get_incoming_response_status = aws_http_stream_get_incoming_response_status,
+    .aws_http_stream_release = aws_http_stream_release,
+    .aws_http_connection_close = aws_http_connection_close};
+
+/*
+ * Tracking structure for each outstanding async query to an sts_web_identity provider
+ */
+struct aws_credentials_provider_sts_web_identity_user_data {
```

Comment:
Nitpick: strictly speaking, since this is not public, it doesn't need the full namespace treatment. If you like, you could just make this ```sts_web_identity_user_data``` or ```sts_web_identity_query``` or something similar/shorter.
Might help shrink some of the insanely long function names :)